### PR TITLE
Corrected explanation of recursive contract creation

### PR DESCRIPTION
**Pull request by mcoblenz**
"now possible" should instead read "not possible."

### CODE REVIEW

**Comment by VoR0220** [comment]
Nope. That's definitely correct phrasing.

**Comment by VoR0220** [comment]
Recommend closing.

**Comment by mcoblenz** [comment]
Then maybe the rest of the sentence could be clearer. "now" is only relevant from a historical perspective, but can only be understood with historical information that is not present. And I don't understand how the second part follows from the first. The first clause introduces a constraint and the second negates a different constraint — what is the relationship there? I was thinking of the case of mutual recursion: if the code of B must be known to instantiate it, and B precedes A, then A can make a new B, but B cannot make a new A unless there is special code in the compiler to tie the knot, right?

**Comment by VoR0220** [comment]
What the docs are saying is that in order for you to create a new instance of a contract from within a contract, that creating contract has to be aware of the source code of the instance of the contract you are trying to create.

**Comment by mcoblenz** [comment]
Ah, I see. According to https://solidity.readthedocs.io/en/develop/contracts.html: "If a contract wants to create another contract, the source code (and the binary) of the created contract has to be known to the creator. This means that cyclic creation dependencies are impossible." This matches what you just wrote in the comment. I've revised the patch accordingly.

File: docs/control-structures.rst [hunk 1]
```diff
@@ -207,8 +207,8 @@ Creating Contracts via ``new``
 ==============================
 
 A contract can create a new contract using the ``new`` keyword. The full
-code of the contract being created has to be known and, thus, recursive
-creation-dependencies are now possible.
+code of the contract being created has to be known; though recursive
```
Comment (pirapira) [inline docs/control-structures.rst]: Maybe `self-recursive` is even clearer.

**Comment by pirapira** [comment]
I don't know why this restriction still exists, but for now the following code does not compile.

```
contract C {
  function f(){
    new D();
  }
}

contract D {
   function d() {
     new C();
  }
}
```

**Comment by mcoblenz** [comment]
@pirapira: interesting. I see this code also does not compile:

```
contract D {
   function d() {
     new D();
  }
}
```

**Comment by pirapira** [comment]
@mcoblenz confirmed.  Now I don't know what are the "recursive" contract creation which is possible.

File: docs/control-structures.rst (hunk 1 above)
Comment (pirapira) [reply docs/control-structures.rst]: No, forget about it.  Self-recursive creation is not possible either.

**Comment by mcoblenz** [comment]
@pirapira @VoR0220 So was my initial suggestion the right one?

**Comment by chriseth** [comment]
Yes, it was.

**Review by chriseth — APPROVED** [review]

**Pull request merged**